content/jss_tracking_commerce.md

```diff
@@ -0,0 +1,171 @@
++++
+title = "JSS Commerce part 4 - Tracking Cart events 🛤️"
+date = "2019-09-19"
+tags = ["XC9", "JSS"]
++++
+
+In [part 3](./jss_cart_actions.md) of our JSS and Commerce series we have been adding cart actions. It would be great if we could track these using Sitecore Experience Analytics.
+This article presents how we achieved tracking a *Lines Added To Cart* event using JSS, and describe the pros and cons of the chosen approach.
+<!--more-->
+
+JSS ships with a tracking API that is pretty well described [here](https://jss.sitecore.com/docs/fundamentals/services/tracking).
+To get started with JSS tracking, we enabled JSS tracking and [tracking of anonymous contacts](https://doc.sitecore.net/developers/xp/xconnect/xconnect-search-indexer/enable-anonymous-contact-indexing.html).
+
+# Perform a page event tracking test
+Next step is to test the out-of-the-box JSS tracking with a test event: 'Download'.
+This is well described on the [documentation](https://jss.sitecore.com/docs/fundamentals/services/tracking) page, but currently it is missing the required fetcher and POST method in its description.
+
+Below code snippet will get you started:
+```
+import { trackingApi } from '@sitecore-jss/sitecore-jss-tracking';
+import { sitecoreApiHost, sitecoreApiKey } from '../temp/config';
+import fetch from 'node-fetch';
+
+const trackingApiOptions = {
+  host: sitecoreApiHost,
+  querystringParams: {
+    sc_apikey: sitecoreApiKey,
+  },
+  fetcher: myFetcher
+};
+
+// Supply your own fetcher, see https://sitecore.stackexchange.com/questions/20929/jss-tracking-api-fetch-cors-and-mvc-errors/20931
+async function myFetcher(url, data) {
+    return await fetch(url, {
+        method: 'post', 
+        headers: {
+            'Content-Type' : 'application/json'
+        }, 
+        body:  JSON.stringify(data)
+    });
+}
+
+trackingApi
+  .trackEvent([{ eventId: 'Download' }], trackingApiOptions)
+  .then(() => console.log('Page event pushed'))
+  .catch((error) => console.error(error));
+```
+
+# Hooking up Commerce events
+Now that we have basic event tracking setup, it's time to hook up a Commerce event.
+As use-case we want to track a *Lines Added To Cart* event. This event is normally triggered by Commerce Connect using the `TriggerCartLinesPageEvent` processor in the `commerce.carts.addCartLines` pipeline.
+
+There are globally 3 places where we could plugin to track an event:
+
+- Commerce Connect pipeline <a id="option-1"></a>
+- Commerce Connect processor
+- IPageContext in Analytics Tracking
+
+First option is to plug in at a Commerce Connect pipeline e.g. `commerce.carts.addCartLines`. This pipeline contains all required actions for adding a cart line, and thus includes the add action for a cart line on the Commerce Engine. This step we do, as described in [part 3](./jss_cart_actions.md), from the client. We could however patch pipeline and remove the unwanted processors.
```

Comment:
Beetje onduidelijk omdat we connect niet gebruiken. Misschien concreet aangeven dat we de pipeline willen gebruiken met alleen de tracking gerelateerde processors